tests/rejected_tx_test.go

```diff
@@ -0,0 +1,128 @@
+	require.NoError(err, "failed to get client")
+
+	chainId := getChainId(t, net)
+
+	testCases := []struct {
+		name    string
+		txMaker func(t *testing.T, account *Account, nonce uint64, price int64) *types.Transaction
+	}{
+		{name: "LegacyTx",
+			txMaker: func(t *testing.T, account *Account, nonce uint64, price int64) *types.Transaction {
+				factory := &txFactory{
+					senderKey: account.PrivateKey,
+					chainId:   chainId,
+				}
+				return factory.makeLegacyTransactionWithPrice(t, nonce, price)
+			},
+		}, {name: "AccessListTx",
+			txMaker: func(t *testing.T, account *Account, nonce uint64, price int64) *types.Transaction {
+				factory := &txFactory{
+					senderKey: account.PrivateKey,
+					chainId:   chainId,
+				}
+				return factory.makeAccessListTransactionWithPrice(t, nonce, price)
+			},
+		}, {name: "DynamicTx",
+			txMaker: func(t *testing.T, account *Account, nonce uint64, price int64) *types.Transaction {
+				factory := &txFactory{
+					senderKey: account.PrivateKey,
+					chainId:   chainId,
+				}
+				return factory.makeDynamicFeeTransactionWithPrice(t, nonce, price)
+			},
+		}, {name: "BlobTx",
+			txMaker: func(t *testing.T, account *Account, nonce uint64, price int64) *types.Transaction {
+				factory := &txFactory{
+					senderKey: account.PrivateKey,
+					chainId:   chainId,
+				}
+				return factory.makeBlobTransactionWithPrice(t, nonce, price)
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			maxFeeCap := getMaxFee(t, client)
+			newAccount := NewAccount()
+			nonce := getNonce(t, client, newAccount.Address())
+			testRejectedTx(t, net, newAccount.Address(), tc.txMaker(t, newAccount, nonce, maxFeeCap))
+		})
+	}
+}
+
+func testRejectedTx(t *testing.T, net *IntegrationTestNet, account common.Address, tx *types.Transaction) {
+	require := require.New(t)
+
+	// verify estimated cost
+	estimatedCost := tx.Gas()*tx.GasFeeCap().Uint64() + tx.Value().Uint64()
+	require.Equal(tx.Cost().Uint64(), estimatedCost, "cost of transaction is not equal to balance")
+
+	// provide just enough balance to NOT cover the cost
+	_, err := net.EndowAccount(account, int64(estimatedCost-1))
+	require.NoError(err, "failed to endow account")
+
+	// run transaction to be rejected
+	receipt, err := net.Run(tx)
+	require.ErrorContains(err, "insufficient funds")
+	require.Nil(receipt)
+
+	// provide enough balance to cover the cost
+	_, err = net.EndowAccount(account, int64(1))
+	require.NoError(err, "failed to endow account")
+
+	// run transaction to be successful
+	receipt, err = net.Run(tx)
+	require.NoError(err)
+
+	// verify receipt
+	require.Equal(receipt.Status, uint64(1))
+	require.Equal(tx.Gas(), receipt.GasUsed)
+	require.GreaterOrEqual(tx.GasFeeCap().Uint64(), receipt.EffectiveGasPrice.Uint64())
+	require.GreaterOrEqual(tx.Cost().Uint64(), receipt.EffectiveGasPrice.Uint64()*receipt.GasUsed)
+}
+
+func getChainId(t *testing.T, net *IntegrationTestNet) *big.Int {
```

Comment:
nit: passing the client instead of the net would eliminate the need to create a client (that should also be closed).